prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -0,0 +1,328 @@
+     * @returns Date of the last time fired
+     */
+    private fun getLastTimeFired( schedule: EmailSchedule ): Date {
+        val parser = CronParser( CronDefinitionBuilder.instanceDefinitionFor(CronType.UNIX) );
+        // Get date for last execution
+        val now = ZonedDateTime.now();
+        val executionTime = ExecutionTime.forCron(parser.parse(schedule.cronSchedule));
+
+        return Date.from(executionTime.lastExecution(now).get().toInstant());
+    }
+
+    /**
+     * Retreives the organization to send the emailschedule to
+     * 
+     * @param schedule the schedule to check organizations against
+     * @returns List of organizations for the schedule
+     */
+    private fun getOrgs( schedule: EmailSchedule ): List<String> {
+        return (
+            if (schedule.organizations.size > 0) 
+                schedule.organizations 
+            else 
+                fetchAllOrgs() 
+        )
+    }
+
+    /**
+     * Retrieves the list of all organization supported
+     * 
+     * @returns List of all organizations supported
+     */
+    private fun fetchAllOrgs(): List<String>{    
+        @Suppress( "NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER" )    
+        var ret = workflowEngine.db.transactReturning {  tx -> 
+            @Suppress( "UNRESOLVED_REFERENCE")
+            val settings = workflowEngine.db.fetchSettings( SettingType.ORGANIZATION, tx )
+            @Suppress( "NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER" )
+            settings.map{ it.getName() }
+        }
+       
+        @Suppress("OVERLOAD_RESOLUTION_AMBIGUITY")
+        System.out.println( ret );
+        return ret;
+    }
+
+
+    @FunctionName("createEmailSchedule")
+    @StorageAccount("AzureWebJobsStorage")
+    fun createEmailSchedule(
+        @HttpTrigger(
+            name = "createEmailSchedule",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-schedule"
+        ) request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        var user:String? = validateUser( request, context );
+        var ret = request.createResponseBuilder(HttpStatus.CREATED);
+
+        if( user !== null ){
+            WorkflowEngine.databaseAccess.insertEmailSchedule(request.body, user );
+        }
+        else{
+            ret.status(HttpStatus.UNAUTHORIZED )
+        }
+        return ret.build();
+    }
+
+
+    fun validateUser( request: HttpRequestMessage<String?>, context: ExecutionContext ): String? {
+        var jwtToken = request.headers["authorization"] ?: ""
+
+        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""
+
+        if (jwtToken.isNotBlank()) {
+            try {
+                // get the access token verifier
+                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                    .setIssuer("https://hhs-prime.okta.com/oauth2/default")
```

Comment:
use constants from `OktaCommands.kt`

Reply:
OktaCommands.kt is in the cli package - but created my own consts at the head of the file